xlators/mgmt/glusterd/src/glusterd-volgen.c

```diff
@@ -2755,6 +2755,114 @@ check_and_add_debug_xl(volgen_graph_t *graph, dict_t *set_dict, char *volname,
     return ret;
 }
 
+gf_boolean_t
+check_user_xlator_position(dict_t *dict, char *key, data_t *value,
+                           void *prev_xlname)
+{
+    if (strncmp(key, "user.xlator.", strlen("user.xlator.")) != 0) {
+        return false;
+    }
+
+    if (fnmatch("user.xlator.*.*", key, 0) == 0) {
+        return false;
+    }
+
+    char *value_str = data_to_str(value);
+    if (!value_str) {
+        return false;
+    }
+
+    if (strcmp(data_to_str(value), prev_xlname) == 0) {
+        gf_log("glusterd", GF_LOG_INFO,
+               "found insert position of user-xlator(%s)", key);
+        return true;
+    }
+
+    return false;
+}
+
+int
+set_user_xlator_option(dict_t *set_dict, char *key, data_t *value, void *data)
+{
+    xlator_t *xl = data;
+    char *optname = strrchr(key, '.') + 1;
+
+    gf_log("glusterd", GF_LOG_DEBUG, "set user xlator option %s = %s", key,
+           value->data);
+
+    return xlator_set_option(xl, optname, strlen(optname), data_to_str(value));
+}
+
+int
+insert_user_xlator_to_graph(dict_t *set_dict, char *key, data_t *value,
+                            void *action_data)
+{
+    int ret = -1;
+
+    struct {
```

Comment:
I don't believe we usually define structs like this in Gluster. We usually leave it in .h files or elsewhere, not within a function.

Reply:
The structure is defined and used in two places. I don't think this is good. Given that it's a private structure, it doesn't need to go to a .h file, but we should move the definition outside of the function. At the top of this file would be a good place.

Reply:
I agree to @xhernandez 's proposal.
So, I'll move the structure to outside of functions.